File: docs/WidgetMill/packages/spec/src/nodes/flexible.ts
import { z } from "zod";
import { FLEX_FIT } from "../enums";
import { NodeRef } from "../node-ref";

/** Flexible — spec §2.12. */
export const FlexibleNode = z.object({
  type: z.literal("flexible"),
  props: z
    .object({
      flex: z.number().int().optional(),
      fit: z.enum(FLEX_FIT).optional(),
    })
    .default({}),
  child: NodeRef.optional(),
});

File: docs/WidgetMill/flutter/sdui_flutter/lib/src/builders/flexible.dart
import 'package:flutter/widgets.dart';
import '../model/sdui_node.dart';
import '../parsing/enums.dart';
import '../renderer.dart';

/// Flexible — spec §2.12.
Widget buildFlexible(BuildContext context, SduiNode node, SduiRenderer r) {
  final p = node.props;
  return Flexible(
    flex: (p['flex'] as num?)?.toInt() ?? 1,
    fit: flexFitFrom(p['fit']),
    child: r.maybeRender(context, node.child) ?? const SizedBox.shrink(),
  );
}
